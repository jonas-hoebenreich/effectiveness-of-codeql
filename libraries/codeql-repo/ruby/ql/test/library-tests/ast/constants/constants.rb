module ModuleA
    class ClassA
        CONST_A = "const_a"
    end

    CONST_B = "const_b"

    module ModuleB
        class ClassB < Base
        end

        class ClassC < X::Y::Z
        end
    end
end

GREETING = 'Hello' + ModuleA::ClassA::CONST_A + ModuleA::CONST_B

def foo
    Names = ['Vera', 'Chuck', 'Dave']

    Names.each do |name|
        puts "#{ GREETING } #{ name }"
    end

    # A call to Kernel::Array; despite beginning with an upper-case character,
    # we don't consider this to be a constant access.
    Array('foo')
end

class ModuleA::ClassD < ModuleA::ClassA
end

module ModuleA::ModuleC
end

ModuleA::ModuleB::MAX_SIZE = 1024

puts ModuleA::ModuleB::MAX_SIZE

puts GREETING
puts ::GREETING
