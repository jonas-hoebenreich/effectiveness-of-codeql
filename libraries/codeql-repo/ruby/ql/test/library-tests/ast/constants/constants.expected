constantAccess
| constants.rb:1:1:15:3 | ModuleA | write | ModuleA | ModuleDeclaration |
| constants.rb:2:5:4:7 | ClassA | write | ClassA | ClassDeclaration |
| constants.rb:3:9:3:15 | CONST_A | write | CONST_A | ConstantAssignment |
| constants.rb:6:5:6:11 | CONST_B | write | CONST_B | ConstantAssignment |
| constants.rb:8:5:14:7 | ModuleB | write | ModuleB | ModuleDeclaration |
| constants.rb:9:9:10:11 | ClassB | write | ClassB | ClassDeclaration |
| constants.rb:9:24:9:27 | Base | read | Base | ConstantReadAccess |
| constants.rb:12:9:13:11 | ClassC | write | ClassC | ClassDeclaration |
| constants.rb:12:24:12:24 | X | read | X | ConstantReadAccess |
| constants.rb:12:24:12:27 | Y | read | Y | ConstantReadAccess |
| constants.rb:12:24:12:30 | Z | read | Z | ConstantReadAccess |
| constants.rb:17:1:17:8 | GREETING | write | GREETING | ConstantAssignment |
| constants.rb:17:22:17:28 | ModuleA | read | ModuleA | ConstantReadAccess |
| constants.rb:17:22:17:36 | ClassA | read | ClassA | ConstantReadAccess |
| constants.rb:17:22:17:45 | CONST_A | read | CONST_A | ConstantReadAccess |
| constants.rb:17:49:17:55 | ModuleA | read | ModuleA | ConstantReadAccess |
| constants.rb:17:49:17:64 | CONST_B | read | CONST_B | ConstantReadAccess |
| constants.rb:20:5:20:9 | Names | write | Names | ConstantAssignment |
| constants.rb:20:13:20:37 | Array | read | Array | ConstantReadAccess |
| constants.rb:22:5:22:9 | Names | read | Names | ConstantReadAccess |
| constants.rb:23:18:23:25 | GREETING | read | GREETING | ConstantReadAccess |
| constants.rb:31:1:32:3 | ClassD | write | ClassD | ClassDeclaration |
| constants.rb:31:7:31:13 | ModuleA | read | ModuleA | ConstantReadAccess |
| constants.rb:31:25:31:31 | ModuleA | read | ModuleA | ConstantReadAccess |
| constants.rb:31:25:31:39 | ClassA | read | ClassA | ConstantReadAccess |
| constants.rb:34:1:35:3 | ModuleC | write | ModuleC | ModuleDeclaration |
| constants.rb:34:8:34:14 | ModuleA | read | ModuleA | ConstantReadAccess |
| constants.rb:37:1:37:7 | ModuleA | read | ModuleA | ConstantReadAccess |
| constants.rb:37:1:37:16 | ModuleB | read | ModuleB | ConstantReadAccess |
| constants.rb:37:1:37:26 | MAX_SIZE | write | MAX_SIZE | ConstantAssignment |
| constants.rb:39:6:39:12 | ModuleA | read | ModuleA | ConstantReadAccess |
| constants.rb:39:6:39:21 | ModuleB | read | ModuleB | ConstantReadAccess |
| constants.rb:39:6:39:31 | MAX_SIZE | read | MAX_SIZE | ConstantReadAccess |
| constants.rb:41:6:41:13 | GREETING | read | GREETING | ConstantReadAccess |
| constants.rb:42:6:42:15 | GREETING | read | GREETING | ConstantReadAccess |
getConst
| constants.rb:1:1:15:3 | ModuleA | CONST_B | constants.rb:6:15:6:23 | "const_b" |
| constants.rb:2:5:4:7 | ModuleA::ClassA | CONST_A | constants.rb:3:19:3:27 | "const_a" |
| file://:0:0:0:0 | Object | GREETING | constants.rb:17:12:17:64 | ... + ... |
lookupConst
| constants.rb:1:1:15:3 | ModuleA | CONST_B | constants.rb:6:15:6:23 | "const_b" |
| constants.rb:2:5:4:7 | ModuleA::ClassA | CONST_A | constants.rb:3:19:3:27 | "const_a" |
| constants.rb:2:5:4:7 | ModuleA::ClassA | GREETING | constants.rb:17:12:17:64 | ... + ... |
| constants.rb:8:5:14:7 | ModuleA::ModuleB | MAX_SIZE | constants.rb:37:30:37:33 | 1024 |
| constants.rb:9:9:10:11 | ModuleA::ModuleB::ClassB | GREETING | constants.rb:17:12:17:64 | ... + ... |
| constants.rb:12:9:13:11 | ModuleA::ModuleB::ClassC | GREETING | constants.rb:17:12:17:64 | ... + ... |
| constants.rb:31:1:32:3 | ModuleA::ClassD | CONST_A | constants.rb:3:19:3:27 | "const_a" |
| constants.rb:31:1:32:3 | ModuleA::ClassD | GREETING | constants.rb:17:12:17:64 | ... + ... |
| file://:0:0:0:0 | Object | GREETING | constants.rb:17:12:17:64 | ... + ... |
constantValue
| constants.rb:17:22:17:45 | CONST_A | constants.rb:3:19:3:27 | "const_a" |
| constants.rb:17:49:17:64 | CONST_B | constants.rb:6:15:6:23 | "const_b" |
| constants.rb:23:18:23:25 | GREETING | constants.rb:17:12:17:64 | ... + ... |
| constants.rb:39:6:39:31 | MAX_SIZE | constants.rb:37:30:37:33 | 1024 |
| constants.rb:41:6:41:13 | GREETING | constants.rb:17:12:17:64 | ... + ... |
| constants.rb:42:6:42:15 | GREETING | constants.rb:17:12:17:64 | ... + ... |
constantWriteAccessQualifiedName
| constants.rb:1:1:15:3 | ModuleA | ModuleA |
| constants.rb:2:5:4:7 | ClassA | ModuleA::ClassA |
| constants.rb:3:9:3:15 | CONST_A | ModuleA::ClassA::CONST_A |
| constants.rb:6:5:6:11 | CONST_B | ModuleA::CONST_B |
| constants.rb:8:5:14:7 | ModuleB | ModuleA::ModuleB |
| constants.rb:9:9:10:11 | ClassB | ModuleA::ModuleB::ClassB |
| constants.rb:12:9:13:11 | ClassC | ModuleA::ModuleB::ClassC |
| constants.rb:17:1:17:8 | GREETING | GREETING |
| constants.rb:20:5:20:9 | Names | Names |
| constants.rb:31:1:32:3 | ClassD | ClassD |
| constants.rb:34:1:35:3 | ModuleC | ModuleC |
| constants.rb:37:1:37:26 | MAX_SIZE | MAX_SIZE |
